### PR TITLE
Fix missing product indexation after creation on product page V2

### LINKED ISSUE #24460: Product V2 page - Indexation of product not done after creation
#### Describe the bug

When creating a product fully on Product V2 page, the product is not automatically indexed. It is correctly indexed in classic Product page.

#### Expected behavior

The page should be indexed when created on Product V2 page

#### Steps to Reproduce

1. Go to Advanced Parameters > Experimental features
2. Enable Experimental Product Page
3. Go to Catalog > Products, click on "New product on experimental page"
4. Fill all needed fields, save the product
5. Go to FO, search for the product : The product is not displayed in the search
6. On another tab, go to BO > Catalog > Products > Edit product on classic product page, and just save the product on classic product page
7. Refresh the search on FO : The product is now displayed in the search

**Screenshots**

If applicable, add screenshots or screen recordings to help explain your problem.

#### Additional information

* PrestaShop version: 1.7.8.x
* PHP version: N/A

### PR DESCRIPTION
**Pull request by matthieu-rolland**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | Fix product not indexed after creation on product page V2 
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #24460
| How to test?      | See steps described in issue #24460
| Possible impacts? | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24774)

### CODE REVIEW

**Review by matks — COMMENTED** [review]
Code seems OK but it would be amazing to have a test to verify indexation is correct after your changes

File: src/Adapter/Product/Update/ProductIndexationUpdater.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+     * @param int $productId
+     */
+    public function updateIndexation(int $productId): void
+    {
+        try {
+            Search::indexation(false, $productId);
+        } catch (PrestaShopException $e) {
+            throw new CoreException(
+                sprintf(
+                    'An error occured while trying to update indexation data for product %s.',
```
Comment (PierreRambaud) [inline src/Adapter/Product/Update/ProductIndexationUpdater.php]:
```suggestion
                    'An error occured while trying to update indexation data for product %d.',
```

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
I agree with @matks Could you please add tests? :heart:

File: src/Adapter/Product/CommandHandler/UpdateProductOptionsHandler.php [hunk 2]
```diff
@@ -50,16 +51,23 @@ final class UpdateProductOptionsHandler implements UpdateProductOptionsHandlerIn
...
     private $manufacturerRepository;
 
+    /**
+     * @var ProductIndexationUpdater
+     */
+    private $productIndexationUpdater;
+
     /**
      * @param ProductRepository $productRepository
      * @param ManufacturerRepository $manufacturerRepository
```
Comment (Progi1984) [inline src/Adapter/Product/CommandHandler/UpdateProductOptionsHandler.php:67]: Please update phpdoc

File: src/Adapter/Product/Update/ProductIndexationUpdater.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Update;
+
+use PrestaShop\PrestaShop\Adapter\CoreException;
```
Comment (Progi1984) [inline src/Adapter/Product/Update/ProductIndexationUpdater.php:31]:
```suggestion
use PrestaShop\PrestaShop\Adapter\CoreException;
use PrestaShopException;
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks & Could you add a behat test, please ?

File: src/Adapter/Product/Update/ProductIndexationUpdater.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+use PrestaShopException
```
Comment (Progi1984) [inline src/Adapter/Product/Update/ProductIndexationUpdater.php]:
```suggestion
use PrestaShopException;
```

**Review by Progi1984 — COMMENTED** [review]
A last bug and it's ok for me

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @matthieu-rolland,

In the BO > Shop Parameters > Search page, we have an option  **Indexing** to enable or disable the indexing.
By default is enabled.
So I have an issue when we disable it following these steps:

1. BO > Shop Parameters > Search page
2. Disable Indexing and save 
image
3. Go to Advanced Parameters > Experimental features
4. Enable Experimental Product Page
5. Go to Catalog > Products, click on "New product on experimental page"
6. Fill all needed fields, save the product
7. Go to FO, search for the product: The product is well displayed in the search

https://drive.google.com/file/d/1rQyGTaIfj3HDQ9aYgBo5ieD4ya2VkH2V/view?usp=sharing

It shouldn't because the indexing feature is disabled

Thank you!

**Review by matks — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @matthieu-rolland,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1FYS_XQ7Qut-44hJ_zjX1S79GlEYvgzKm/view?usp=sharing

Thank you!

**Comment by Progi1984** [comment]
Thanks @matthieu-rolland & @khouloudbelguith

**Pull request merged**